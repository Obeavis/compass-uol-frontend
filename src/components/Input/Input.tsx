import React, { FC, InputHTMLAttributes } from "react";

interface Props extends InputHTMLAttributes<HTMLInputElement> {
  name: string;
  error?: boolean;
  icon?: React.ReactNode;
  className?: string;
}

const Input: FC<Props> = ({ name, error, icon, className, ...rest }) => {
  return (
    <label
      className={`w-full flex items-center gap-3 ${className}`}
      htmlFor={name}
    >
      <input
        className={`rounded-full border bg-transparent text-white w-full px-3 py-2 ${
          error ? "border-secondary" : "border-gray-200"
        }`}
        id={name}
        {...rest}
      />
      {icon}
    </label>
  );
};

export default Input;
