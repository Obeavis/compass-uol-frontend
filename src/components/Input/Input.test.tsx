import { render } from "@testing-library/react";
import Input from "./Input";

describe("Button component", () => {
  test("renders the component", async () => {
    const { container } = render(<Input name="teste" />);

    expect(container).toMatchSnapshot();
  });
});
