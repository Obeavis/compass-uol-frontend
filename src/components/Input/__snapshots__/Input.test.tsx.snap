// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders the component 1`] = `
<div>
  <label
    class="w-full flex items-center gap-3 undefined"
    for="teste"
  >
    <input
      class="rounded-full border bg-transparent text-white w-full px-3 py-2 border-gray-200"
      id="teste"
    />
  </label>
</div>
`;
